patch_ts_handlers.py

```
@@ -0,0 +1,78 @@
+    for action in actions_to_patch:
+        # Check if already sanitized
+        check_pattern = rf"case '{action}': {{.*?sanitizePath\(argsRecord\.widgetPath.*?"
+        if re.search(check_pattern, content, flags=re.DOTALL):
+            continue # Already patched this one
+
+        pattern = rf"(case '{action}': {{\s*requireNonEmptyString\(argsRecord\.widgetPath, 'widgetPath', 'Missing required parameter: widgetPath'\);(.*?)\s*return sendRequest\('{action}'\);\s*}})"
+        
+        def repl(m):
+            inner = m.group(2)
+            new_inner = inner + "\n      argsRecord.widgetPath = sanitizePath(argsRecord.widgetPath as string);\n"
+            return f"case '{action}': {{\n      requireNonEmptyString(argsRecord.widgetPath, 'widgetPath', 'Missing required parameter: widgetPath');{new_inner}      return sendRequest('{action}');\n    }}"
```

Ruff (0.15.15)

[warning] Function definition does not bind loop variable `action`

(B023)